src/test/java/io/vlingo/http/resource/sse/MockResponseSenderChannel.java

```diff
@@ -21,20 +21,51 @@
   public AtomicInteger abandonCount = new AtomicInteger(0);
   public AtomicInteger respondWithCount = new AtomicInteger(0);
   public AtomicReference<Response> response = new AtomicReference<>();
-  public TestUntil untilAbandon;
-  public TestUntil untilRespondWith;
+  private AccessSafely abandonSafely;
+  private AccessSafely respondWithSafely;
 
   @Override
   public void abandon(final RequestResponseContext<?> context) {
     abandonCount.incrementAndGet();
-    if (untilAbandon != null) untilAbandon.happened();
+    if (abandonSafely != null) {
+      abandonSafely.writeUsing("foo", "bar");
+    }
   }
 
   @Override
   public void respondWith(final RequestResponseContext<?> context, final ConsumerByteBuffer buffer) {
     final ResponseParser parser = ResponseParser.parserFor(buffer.asByteBuffer());
     response.set(parser.fullResponse());
     respondWithCount.incrementAndGet();
-    if (untilRespondWith != null) untilRespondWith.happened();
+    if (respondWithSafely != null) {
+      respondWithSafely.writeUsing("foo", "bar");
+    }
   }
+
+  public void expectAbandon(int n) {
+    if (abandonSafely != null) {
+      throw new IllegalStateException("call to untilAbandon without corresponding call to completes");
+    }
+    abandonSafely = AccessSafely.afterCompleting(n).writingWith("foo", (x) -> {}).readingWith("foo", () -> "bar");
+  }
+  
+
+  public void expectRespondWith(int n) {
```

Comment:
Answer the new `AccessSafely` instance in `respondWithSafely` so it can be used in test assertions.